File: Broker.UnitTests/BrokerMessageTests.cs
using System;
using NUnit.Framework;
using Psns.Common.Clients.Broker;

namespace Broker.UnitTests
{
    [TestFixture]
    public class BrokerMessageTests : AssertionHelper
    {
        [Test]
        public void Constructor_MessageAsNull_MessageSetToEmptyString()
        {
            Expect(new BrokerMessage("Contract", "type", null, Guid.Empty, Guid.Empty).Message, EqualTo(string.Empty));
        }

        [Test]
        public void Constructor_MessageTypeAsNull_MessageTypeSetToEmptyString()
        {
            Expect(new BrokerMessage("Contract", null, "Message", Guid.Empty, Guid.Empty).MessageType, EqualTo(string.Empty));
        }

        [Test]
        public void Equals_BothEmpty_AreEqual()
        {
            Expect(BrokerMessage.Empty, EqualTo(BrokerMessage.Empty));
            Expect(BrokerMessage.Empty == new BrokerMessage(string.Empty, string.Empty, string.Empty, Guid.Empty, Guid.Empty), Is.True);
        }

        [Test]
        public void Equals_BothDifferent_AreNotEqual()
        {
            Expect(BrokerMessage.Empty != new BrokerMessage("contract", "type", "", Guid.Empty, Guid.Empty), Is.True);
        }
    }
}

File: Broker.FUnitTests/BrokerMessageTests.fs
module BrokerMessageTests

open System
open System.Xml.Linq
open System.Text.RegularExpressions
open NUnit.Framework
open FsUnit
open Psns.Common.Clients.Broker
open Psns.Common.Functional

type ex = Psns.Common.Functional.Prelude
let xname str = XName.op_Implicit str
let doTry (t: Try<'a>) = t.Match((fun x -> x), (fun ex -> failwith ex.Message))
let removeSpaces str = Regex.Replace(str, @"\s+", "")

[<Test>]
let ``it should serialize properly.`` () =
    let xml = new XElement(xname "root", new XElement(xname "child", "value"))

    let json = doTry ((new BrokerMessage ("contract", "type", removeSpaces (xml.ToString()), Guid.Empty, Guid.Empty)).AsJson())
    let message = doTry (json.FromJson())

    Seq.iter
        (fun pair -> fst pair |> should equal (snd pair))
        ["contract", message.Contract; "type", message.MessageType; "<root><child>value</child></root>", message.Message]